### ISSUE TITLE
[lint request] lifetime_extending_transmute

### ISSUE THREAD

**Issue by gnzlbg** [rust-lang/rust-clippy#4000]
Some people use "identity" transmutes to arbitrarily extend lifetimes, e.g.,

```rust
fn foo<'a>(x: &'a i32) -> &'static i32 {
    unsafe { std::mem::transmute(x) }
}
```

We should warn on these. Sure, they aren't necessarily incorrect, but they are definitely suspicious.
